host/src/processgrid.cpp

```diff
@@ -167,18 +177,20 @@ int get_gridvalues_f(const Gridinfo* mygrid, float* fgrids, bool cgmaps)
 	for (t=0; t < mygrid->num_of_atypes+2; t++)
 	{
 		//opening corresponding .map file
-		//-------------------------------------
-		// Added the complete path of associated grid files.
-		strcpy(tempstr,mygrid->grid_file_path);
-		strcat(tempstr, "/");
-		strcat(tempstr, mygrid->receptor_name);
-		
-		//strcpy(tempstr, mygrid->receptor_name);
-		//-------------------------------------
+		strcpy(tempstr,mygrid->map_base_name);
 		strcat(tempstr, ".");
 		strcat(tempstr, mygrid->grid_types[t]);
 		strcat(tempstr, ".map");
 		fp = fopen(tempstr, "rb"); // fp = fopen(tempstr, "r");
+		if (fp == NULL){ // try again with the receptor name in the .maps.fld file
+			strcpy(tempstr,mygrid->grid_file_path);
+			strcat(tempstr, "/");
+			strcat(tempstr, mygrid->receptor_name);
+			strcat(tempstr, ".");
+			strcat(tempstr, mygrid->grid_types[t]);
+			strcat(tempstr, ".map");
+			fp = fopen(tempstr, "rb"); // fp = fopen(tempstr, "r");
+		}
 		if (fp == NULL)
```

Comment:
@atillack: Line 194 evaluates the exact same condition as above (line 185). Either these two conditional blocks should be merged or ... if I understand the intent correctly, it would be better to:
```
...
open using map_base_name
if (fp = NULL) {
  instead open using receptor name
}
else {
  printf("Error: can't open %s!\n", tempstr);
  ...
}
...
```

Reply:
One is a fallback of the other - if the first file can't be opened, it goes inside the condition and tries to open the file specified in the grid file, if that fails the second condition throws an error. In case the first one succeeds, both succeed.

Reply:
Ah OK, I missed the fact that one is fallback of the other. So, you are right, they must not be merged.

Now, for a clearer implementation, shouldn't the block comprised by the second call to `if (fp == NULL){}`  placed right after the second `fp = fopen(tempstr, "rb");`  and enclosed within the first `if (fp == NULL){}` ? So something like this instead:

```
fp = fopen(tempstr, "rb");
if (fp == NULL){ // try again with the receptor name in the .maps.fld file
  ...
  fp = fopen(tempstr, "rb"); 
  if (fp == NULL){
    printf("Error: can't open %s!\n", tempstr);
    ...
    return 1;
  }
}

//seeking to first data
...
```

Reply:
After an internal discussion with @atillack, I realized it makes sense to keep the proposed implementation (decoupled `if (fp == NULL)` blocks ) for readability and maintanability.